storage/cloud-client/hmac.py

```diff
@@ -0,0 +1,154 @@
+                                project_id=project_id)
+    print('The base64 encoded secret is {}'.format(secret))
+    print('Do not miss that secret, there is no API to recover it.')
+    print('The HMAC key metadata is:')
+    print('Key ID: {}'.format(hmac_key.id))
+    print('Access ID: {}'.format(hmac_key.access_id))
+    print('Project ID: {}'.format(hmac_key.project))
+    print('State: {}'.format(hmac_key.state))
+    print('Created At: {}'.format(hmac_key.time_created))
+    print('Updated At: {}'.format(hmac_key.updated))
+    print('Etag: {}'.format(hmac_key.etag))
+    # [END storage_create_hmac_key]
+    return hmac_key
+
+
+def get_key(access_id, project_id):
+    """
+    Retrieve the HMACKeyMetadata with the given access id.
+    """
+    # [START storage_get_hmac_key]
+    # project_id = 'Your Google Cloud project ID'
+    # access_id = 'ID of an HMAC key'
+    storage_client = storage.Client(project=project_id)
+    hmac_key = storage_client.get_hmac_key_metadata(access_id,
+                                                    project_id=project_id)
+    print('The HMAC key metadata is:')
+    print('Key ID: {}'.format(hmac_key.id))
+    print('Access ID: {}'.format(hmac_key.access_id))
+    print('Project ID: {}'.format(hmac_key.project))
+    print('State: {}'.format(hmac_key.state))
+    print('Created At: {}'.format(hmac_key.time_created))
+    print('Updated At: {}'.format(hmac_key.updated))
+    print('Etag: {}'.format(hmac_key.etag))
+    # [END storage_get_hmac_key]
+    return hmac_key
+
+
+def activate_key(access_id, project_id):
+    """
+    Activate the HMAC key with the given access ID.
+    """
+    # [START storage_activate_hmac_key]
+    # project_id = 'Your Google Cloud project ID'
+    # access_id = 'ID of an inactive HMAC key'
+    storage_client = storage.Client(project=project_id)
+    hmac_key = storage_client.get_hmac_key_metadata(access_id,
+                                                    project_id=project_id)
+    hmac_key.state = 'ACTIVE'
+    hmac_key.update()
+    print('The HMAC key metadata is:')
+    print('Key ID: {}'.format(hmac_key.id))
+    print('Access ID: {}'.format(hmac_key.access_id))
+    print('Project ID: {}'.format(hmac_key.project))
+    print('State: {}'.format(hmac_key.state))
+    print('Created At: {}'.format(hmac_key.time_created))
+    print('Updated At: {}'.format(hmac_key.updated))
+    print('Etag: {}'.format(hmac_key.etag))
+    # [END storage_activate_hmac_key]
+    return hmac_key
+
+
+def deactivate_key(access_id, project_id):
+    """
+    Deactivate the HMAC key with the given access ID.
+    """
+    # [START storage_deactivate_hmac_key]
+    # project_id = 'Your Google Cloud project ID'
+    # access_id = 'ID of an active HMAC key'
+    storage_client = storage.Client(project=project_id)
+    hmac_key = storage_client.get_hmac_key_metadata(access_id,
+                                                    project_id=project_id)
+    hmac_key.state = 'INACTIVE'
```

Comment:
What's the significance of this value? Is it always `'INACTIVE'` or `'ACTIVE'` -- does the library provide an ENUM for these values? When should I use which -- perhaps link to a docs page with more context?

Reply:
It's from here: https://github.com/googleapis/google-cloud-python/blob/master/storage/google/cloud/storage/hmac_key.py#L33 . There's no enum in the client library for this (though I think I would have used one probably...).

The canonical sample seems to use the same pattern: https://github.com/googleapis/nodejs-storage/blob/master/samples/hmacKeyActivate.js#L44 . We could add a docs link across all samples if it seems necessary (I don't know which docs should be linked to but I can look into it).

Reply:
I also considered doing this but I'm a little concerned that this will be more confusing to whoever is reading/using the samples. The current version works with just one import (`from google.cloud import storage`) ; in order to use the class attributes the user would have to add `from google.cloud.storage.hmac_key import HMACKeyMetadata` as well. I can go this route if you prefer though.

Reply:
Ok, that seems reasonable. I think if we were using the newer style samples where the imports are included it would make more sense, but since in this version imports out of the snippet I can see how it might be more confusing.